### ISSUE TITLE
Test.pm: regression caused by the Perl abs2rel workaround

### ISSUE THREAD

**Issue by mspncp** [openssl/openssl#10628]
While working on https://github.com/openssl/openssl/issues/10604#issuecomment-564458060, I noticed that the tests now behave differently from what I was accustomed to: all paths in the test commands are given as absolute paths:

```
~/src/openssl$ make V=1 TESTS=test_ocsp tests
...
/home/msp/src/openssl/util/shlib_wrap.sh /home/msp/src/openssl/apps/openssl base64 -d -in /home/msp/src/openssl/test/ocsp-tests/ND1.ors -out ocsp-resp-fff.dat => 0
```

Until recently, the paths used to be relative to the test directory:

```
~/src/openssl$ make V=1 TESTS=test_ocsp tests
...
../../util/shlib_wrap.sh ../../apps/openssl base64 -d -in ../../test/ocsp-tests/ND1.ors -out ocsp-resp-fff.dat => 0
```

This was very handy for several reasons: For example, it was easy to copy & paste a command from a [verbose test run](https://github.com/openssl/openssl/issues/10604#issuecomment-564458060) of someone else into your shell (or from your master worktree into your 1.1.1. worktree) and it could be **run without editing**. If necessary, the command could also be debugged by inserting in `gdb --args` in the right location:

```
~/src/openssl$ make V=1 TESTS=test_ocsp tests
[...]
../../util/shlib_wrap.sh gdb --args ../../apps/openssl base64 -d -in ../../test/ocsp-tests/ND1.ors -out ocsp-resp-fff.dat => 0
```

The cause for this behavioral change is the following commit which added a workaraund for a Perl abs2rel bug:

* 7a2027240e1d01f7f5b209998d1de36af221b34b Adding Test.pm with workaround for Perl abs2rel bug [master]
* 374cab6390ba005d4a559a3dea2a034af9cb1c09 Adding Test.pm with workaround for Perl abs2rel bug [1.1.1]

I consider this behavioral change a regression, because it impedes the process of analyzing test failures. That's why I tagged it as a bug and I would appreciate very much if @mpeylo and @levitte would find an alternative workaround which does not have this negative side effect.

**Comment by mspncp** [comment]
@richsalz, since @levitte is very busy at the moment, maybe you could do me the favour and take a look at this issue when you have some spare time?

**Comment by richsalz** [comment]
I will take a look.  This is beyond my level of Perl skill, so no promises.

**Comment by richsalz** [comment]
Perhaps Test.pm can write a small testcase in Test.pm and, if the bug appears, use the abspath function and if the bug does not appear use the old behavior?  @mpeylo ?

**Comment by levitte** [comment]
That's an idea. "Steal" the fix that was added in File::Spec kinda thing?

**Comment by richsalz** [comment]
Yeah.  I'd write the test, but I don't understand the text in the commit well enough to do so.

**Comment by levitte** [comment]
So if I remember this correctly, the whole issue was when `abs2rel` was called with relative paths, not when it was called with absolute paths, and somewhere deep down, this was because of some bug in the way it found the current directory (it used an XS method which turned out to have bugs, if I understand correctly, and more recent versions of File::Spec::Unix use the pure perl `Cwd::getcwd` instead).

Sooooo, I can only conclude that and abs2rel in `__srctop_file` and friends, with `getcwd` as base, should be safe to do.

**Comment by levitte** [comment]
Fix in #10913

**Comment by mspncp** [comment]
Thank you. I'll take a look as soon as possible.
